debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/view/dtfj/java/j9/HeapObjectIterator.java

```diff
@@ -57,6 +58,21 @@ public void doDeadObject(J9ObjectPointer object) {
 	}
 
 	public void doLiveObject(J9ObjectPointer ptr) {
+		try {
+			if (J9ObjectHelper.clazz(ptr).classObject().getAddress() == -1) {
```

Comment:
This will not work 32-bit dumps where the all-ones pointer value would be `0xFFFFFFFFL`.
Also, this is the wrong place for this test: it belongs in `GCObjectModel_V1.isDeadObject(J9ObjectPointer)`.

Reply:
> This will not work for 32-bit dumps

Makes sense. Do I check `J9BuildFlags.env_data64`?

> it belongs in GCObjectModel_V1.isDeadObject(J9ObjectPointer).

That's actually where I first put the patch but I encountered some very strange behavior where marking these objects as dead caused a drastic change in the number of total live objects from 1,427,358 to 113,626, whereas placing the check in `HeapObjectIterator` knocked down only the expected amount (to 1,427,353). Might there be some logic along the lines of: if we find a dead object in this section, then assume the rest of the section is dead? Anyway, I'll take a look...

Reply:
> Do I check `J9BuildFlags.env_data64`

At least indirectly, yes. I suggest adding:
```
    private static final J9ObjectPointer J9_INVALID_OBJECT = J9ObjectPointer.cast(~0L);
```
and changing the test to `J9_INVALID_OBJECT.eq(J9ObjectHelper.clazz(object).classObject())`.

Reply:
I think I figured out the issue: The heap iterator checks for a dead object and advances the search pointer by `ObjectModel.getSizeInBytesDeadObject` so I think I just need to add the dark matter check in `getSizeInBytesDeadObject` to instead return `ObjectModel.getConsumedSizeInBytesWithHeader`.

Thanks Dmitri, I think I'll change to `J9AccClassDying` as it's more logical.